File: web/src/api/stats.ts
import type { GlobalStats, ApiResponse } from '../types'

export async function getStats(): Promise<GlobalStats> {
  const res = await fetch('/api/v1/stats')
  const json: ApiResponse<GlobalStats> = await res.json()
  if (!json.success) throw new Error(json.error)
  return json.data!
}

File: crates/ferroload-api/src/routes/stats.rs
use actix_web::{web, HttpResponse};
use serde_json::json;
use crate::state::AppState;

pub async fn get_stats(state: web::Data<AppState>) -> HttpResponse {
    let stats = state.engine.global_stats().await;
    HttpResponse::Ok().json(json!({ "success": true, "data": stats }))
}
